benchmarks/dynamo/common.py

```diff
@@ -1250,78 +1257,59 @@ def wrapper(self, *args, **kwargs) -> Any:
     return wrapper
 
 
-class OnnxModelFromTorchScript:
-    """TorchScript based onnx export. `torch.onnx.export`
-
-    TODO(bowbao):
-    * large model export failed.
-          Onnx Model is larger than 2GB, but exporter makes decision based pt model size, which is
-          smaller than 2GB.
-    * OOM on slightly larger model.
-          Both pt model and ort inference session are on gpu. Attempt has been made to move ORT to
-          cuda:1, however ORT perf drop significantly.
-          For now running everything with batch_size 1 set in launch script.
-    """
-
-    TORCH_TO_NUMPY_DTYPE = {
-        torch.float16: np.float16,
-        torch.float32: np.float32,
-        torch.float64: np.float64,
-        torch.uint8: np.uint8,
-        torch.int8: np.int8,
-        torch.int16: np.int16,
-        torch.int32: np.int32,
-        torch.int64: np.longlong,
-        torch.bool: np.bool_,
-    }
+class OnnxModel(abc.ABC):
+    _COMPILER_NAME: str
 
     def __init__(self, output_directory, model, example_inputs, dynamic_shapes: bool):
-        assert not dynamic_shapes, "NYI dynamic shapes for OnnxModelFromTorchScript"
-        self.model_path = self._generate_onnx_model_path(output_directory)
-        self._export(
-            model,
-            example_inputs,
-            self.model_path,
-            opset_version=17,
-            do_constant_folding=False,
-            verbose=False,
-        )
-        self.onnx_session = self._init_ort_session(self.model_path)
-
-    def _generate_onnx_model_path(
-        self, output_directory: str, onnx_model_folder_name: str = "bench_onnx_models"
-    ) -> str:
         # Hack to get model name.
         from torch._functorch import aot_autograd
 
         model_name = aot_autograd.model_name
-        model_path = pathlib.Path(output_directory, onnx_model_folder_name, model_name)
+        self.model_dir = self._generate_onnx_model_directory(
+            output_directory, self._COMPILER_NAME, model_name
+        )
+        self.model_path = str(
```

Comment:
```suggestion
        self.model_path = os.fspath(
```

nit: consider https://docs.python.org/3/library/os.html#os.fspath

Reply:
isn't str() the recommended way of converting Path to string?

Reply:
I asked sometime in the Python channel and was recommended to use os.fspath. If the path can be btypes, we should use os.fsdecode